src/io/flutter/actions/OpenInXcodeAction.java

```diff
@@ -72,7 +76,43 @@ private VirtualFile findProjectFile(@Nullable AnActionEvent e) {
   }
 
   private static void openFile(@NotNull VirtualFile file) {
-    final String path = file.getPath();
+    final Project project = ProjectUtil.guessProjectForFile(file);
+    final FlutterSdk sdk = project != null ? FlutterSdk.getFlutterSdk(project) : null;
+    if (sdk == null) {
+      FlutterSdkAction.showMissingSdkDialog(project);
+      return;
+    }
+
+    final PubRoot pubRoot = PubRoot.forFile(file);
+    if (pubRoot == null) {
+      FlutterMessages.showError("Error Opening Xcode", "Unable to run `flutter build` (no pub root found)");
+      return;
+    }
+
+    // Trigger an iOS build if necessary.
+    if (!hasBeenBuilt(pubRoot)) {
+      final ProgressHelper progressHelper = new ProgressHelper(project);
+      progressHelper.start("Building for iOS");
+
+      sdk.flutterBuild(pubRoot, "ios", "--debug").start(null, new ProcessAdapter() {
+        @Override
+        public void processTerminated(@NotNull ProcessEvent event) {
+          super.processTerminated(event);
+          progressHelper.done();
+          openWithXcode(file.getPath());
```

Comment:
We should probably check `event.getExitCode()` here, and display an error if there was a build failure (else open Xcode).

Reply:
👍